### PR TITLE
Clean up battery, alert and hidpp related code

### PR DESCRIPTION
**Pull request by MattHag**
Simplifies hierarchy, adds type hints and enhances readability. Fixes capitalization of constants, which is violated by NamedInts.

Related #2273

### CODE REVIEW

**Comment by MattHag** [comment]
Don't know what's the issue with the more or less unrelated test failure

**Review by pfps — COMMENTED** [review]
This commit looks fine.

**Comment by pfps** [comment]
The third commit looks fine.

**Comment by pfps** [comment]
I do not understand the rationale for creating fake classes for a small part of Device.

**Comment by MattHag** [comment]
Which fake classes?

**Comment by pfps** [comment]
Can IntEnum be used instead of NamedInt everywhere in Solaar?   If so, then it is a good idea to do the replacement.   If not, then Solaar should stick with NamedInt throughout.

The NamedInts that need to be checked are the ones that have default values, I think.

**Comment by pfps** [comment]
DeviceProtocol

**Comment by MattHag** [comment]
> Can IntEnum be used instead of NamedInt everywhere in Solaar?
Sure, here it was useless to have special implementation, loose proper type hints and linter checks.

The protocol makes clear which functions are expected and are the foundtion for further decoupling and boring, effective tests and implementations.

**Comment by MattHag** [comment]
Rebased and fixed what I am aware of.

**Comment by MattHag** [comment]
@pfps ready

**Comment by pfps** [comment]
I am not convinced of the need to add the protocols.   If there is a PR without these commits, I'll merge it in.

**Comment by MattHag** [comment]
The protocols are important for further clean up of the data read/write related code, as they indicate, which actions are needed in a module. The hidpp, device and base modules have interesting data paths and thus dependencies, which can be tackled on top of that.

**Comment by pfps** [comment]
If the prototols are important for future work split them off for now.  When the future work is being performed they can be re-evaluated.

**Comment by MattHag** [comment]
Fixed and simplified

**Comment by pfps** [comment]
I'm seeing extra messages

```
2024-05-11 11:50:34,378,378     INFO [MainThread] hidapi.udev: Report Descriptor not processed for DEVICE /dev/hidraw0 BID 0003 VID 0000046D PID 0000C52B: open.<locals>.matchfn() takes 3 positional arguments but 5 were given
```

File: lib/solaar/ui/diversion_rules.py [hunk 1]
```diff
@@ -1156,17 +1166,17 @@ def __init__(self, on_change, *args, accept_toggle=True, **kwargs):
             self.pack_end(w, True, True, 0)
             w.hide()
         self.unsupp_value = None
-        self.current_kind = None
+        self.current_kind: SetValueControlKinds = Optional[None]
```
Comment (MattHag) [inline lib/solaar/ui/diversion_rules.py:1169]: None

File: lib/logitech_receiver/common.py [hunk 2]
```diff
@@ -550,63 +541,81 @@ def __getattr__(self, k):
...
-    level: Optional[Union[NamedInt, int]]
+    ATTENTION_LEVEL = 5
+
+    level: Optional[Union[BatteryLevelApproximation, int]]
     next_level: Optional[Union[NamedInt, int]]
-    status: Optional[NamedInt]
+    status: Optional[BatteryStatus]
     voltage: Optional[int]
-    light_level: Optional[int] = None  # light level for devices with solaar recharging
+    light_level: Optional[int] = None  # light level for devices with solaar RECHARGING
```
Comment (MattHag) [inline lib/logitech_receiver/common.py:580]: Lowercase

**Comment by pfps** [comment]
@MattHag Is there a better way to set up the Linux vs MacOS hid code?  Would it be a good idea to move the hidapi directory to be a subdirectory of logitech_receiver?

**Comment by MattHag** [comment]
Yes, there's certainly a different one, but it's not the most relevant.

No, all those low level communication interface should be self-contained and have no dependency or tight connection to the main Solaar code `/solaar`. 
The HID package should probably be used like a pip dependency and all Solaar specific code from there should be moved to the solaar package, as described in #2480. The addition of #2481 will make it easier to talk about the purpose of the packages and reason about improvements.

**Comment by pfps** [comment]
The problem is that currently Solaar installs hidapi as a top-level module in .../site-packages and there is a separate hidapi package in PyPI.    So either Solaar should just use that package or the Solaar HID code should be moved inside the low-level part of Solaar, i.e., inside logitech_receiver.  The question is which way to go and if the HID code is retained how best to interact with the two back-ends.

At the same time, it is probably a good idea to rename logitech_receiver as it isn't just about receivers.

**Comment by MattHag** [comment]
> I'm seeing extra messages when running `bin/solaar -dd` and a device not showing up.
> 
> ```
> 2024-05-11 11:50:34,378,378     INFO [MainThread] hidapi.udev: Report Descriptor not processed for DEVICE /dev/hidraw0 BID 0003 VID 0000046D PID 0000C52B: open.<locals>.matchfn() takes 3 positional arguments but 5 were given
> ```

Now it will work again.

**Review by pfps — APPROVED** [review]
Looks acceptable

**Pull request merged**